test/function-test.js

```diff
@@ -1,35 +1,114 @@
 /** @license MIT License (c) copyright 2016 original author or authors */
 
-import {describe, it} from 'mocha';
-import assert from 'assert';
+import {describe, it} from 'mocha'
+import assert from 'assert'
 
-import { id, compose, apply } from '../src/function';
+import { id, compose, apply, curry2, curry3 } from '../src/function'
 
 describe('id', () => {
   it('id(x) === x', () => {
-    const x = {};
-    assert.strictEqual(id(x), x);
-  });
-});
+    const x = {}
+    assert.strictEqual(id(x), x)
+  })
+})
 
 describe('compose', () => {
   it('compose(f, g)(x) === f(g(x))', () => {
-    const fx = '' + Math.random();
-    const gx = '' + Math.random();
-    const x = '' + Math.random();
+    const fx = '' + Math.random()
+    const gx = '' + Math.random()
+    const x = '' + Math.random()
 
-    const f = x => x + fx;
-    const g = x => x + gx;
-    const h = compose(f, g);
+    const f = x => x + fx
+    const g = x => x + gx
+    const h = compose(f, g)
 
-    assert.strictEqual(f(g(x)), h(x));
-  });
-});
+    assert.strictEqual(f(g(x)), h(x))
+  })
+})
 
 describe('apply', () => {
   it('apply(f, x) === f(x)', () => {
-    const x = Math.random();
-    const f = x => x+1;
-    assert.strictEqual(apply(f, x), f(x));
-  });
-});
+    const x = Math.random()
+    const f = x => x + 1
+    assert.strictEqual(apply(f, x), f(x))
+  })
+})
+
+describe('curry2', () => {
+  it('should return the original function if no args are given', () => {
+    const fn = (a, b) => a + b
+    const curriedFn = curry2(fn)
+    assert.strictEqual(curriedFn().length, 2)
+  })
+
+  it('should return a function of length 1 when 1 arg is given', () => {
+    const fn = (a, b) => a + b
+    const curriedFn = curry2(fn)
+    assert.strictEqual(curriedFn(1).length, 1)
+  })
+
+  it('should execute function when given 2 args', () => {
+    const fn = (a, b) => a + b
+    const curriedFn = curry2(fn)
+    assert.strictEqual(typeof curriedFn(1, 2).length, 'undefined')
+    assert.strictEqual(curriedFn(1, 2), 3)
+  })
+
+  it('should be invariant', () => {
+    const f = (x, y) => x + y
+    const a = 1
+    const b = 2
+
+    const bool =
+      curry2(f)(a)(b) === 3 &&
+      curry2(f)(a, b) === 3 &&
+      f(a, b) === 3
+
+    assert.strictEqual(bool, true)
```

Comment:
Any of these is fine w/me.  Just trying to see which might be the most readable.